wp_api/src/request/endpoint.rs

```diff
@@ -121,16 +121,22 @@ trait UrlExtension {
 }
 
 impl UrlExtension for Url {
-    fn append(mut self, segment: &str) -> Result<Url, ()> {
-        self.path_segments_mut()?.push(segment);
-        Ok(self)
+    fn append(self, segment: &str) -> Result<Url, ()> {
+        self.extend([segment])
     }
 
     fn extend<I>(mut self, segments: I) -> Result<Url, ()>
```

Comment:
BTW, is there any reason why `UrlExtension` functions take `self` instead of `&self`?

Reply:
Since the function return `Result<Url>`, does that mean the function can be declared as `fn append(&self, ...) -> Result<Url, ()>`, which looks more similar to other Rust code?

Reply:
If you do that, you'll have to clone the value because you can't modify the existing one. When we use `fn append(self)`, the original value is moved to the function so we are free to modify it and then we return the original value.

Keep in mind that these are compile time semantics for Rust. In practice, we are modifying the object in place just like we'd do with `&mut self` so the memory address doesn't change.